lib/migration.js

```diff
@@ -37,6 +37,35 @@ function mixinMigration(MySQL, mysql) {
     });
   };
 
+  MySQL.prototype.getConstraintTrigger = function(model, actualFks, cb) {
+    var table = this.tableEscaped(model);
+    var sql = 'SHOW CREATE TABLE ' + table;
+    this.execute(sql, function(err, createTable) {
+      if (err) {
+        return cb(err);
+      } else {
+        var matchConstraint = new RegExp('CONSTRAINT `([^`]+)` FOREIGN KEY \\(`([^`]+)`\\)' +
+          ' REFERENCES `([^`]+)` \\(`([^`]+)`\\)' +
+          '(?: ON DELETE (RESTRICT|CASCADE|SET NULL|NO ACTION|SET DEFAULT))?' +
+          '(?: ON UPDATE (RESTRICT|CASCADE|SET NULL|NO ACTION|SET DEFAULT))?', 'g');
+        var rawConstraints = [];
+        do {
+          var match = matchConstraint.exec(createTable[0]['Create Table']);
+          if (match) {
+            actualFks.forEach(function(fk) {
+              if (fk.fkName === match[1]) {
+                fk.onDelete = match[5] ? match[5].toLowerCase() : 'restrict';
+                fk.onUpdate = match[6] ? match[6].toLowerCase() : 'restrict';
+              }
+            });
+            rawConstraints.push(match);
+          }
+        } while (match != null);
```

Comment:
the `while` loop is a little bit risky to me, what if the statement doesn't contain the constraint? Maybe change the logic to stop when reaches the last line of the statement.

Reply:
the `if` statement line 54 in the `while` protect for no match in the `CREATE TABLE`, I think the tricky part is when someone rework my code later, because the iteration over the Regex is not obvious. Maybe add some comment before the line 52 to explain the iteration on the multiple match with RegExp object internal pointer.
Or is there a other way to iterate with complex RegExp ?